src/EnergyPlus/HWBaseboardRadiator.cc

```diff
@@ -1557,41 +1430,20 @@ namespace HWBaseboardRadiator {
         //       DATE WRITTEN   Nov 1997
         //                      February 2001
         //       MODIFIED       Aug 2007 Daeho Kang (Add the update of radiant source)
-        //       RE-ENGINEERED  na
-
-        // PURPOSE OF THIS SUBROUTINE:
 
         // METHODOLOGY EMPLOYED:
         // The update subrotines both in high temperature radiant radiator
         // and convective only baseboard radiator are combined and modified.
 
-        // REFERENCES:
-        // na
-
-        // Using/Aliasing
-        using PlantUtilities::SafeCopyPlantNode;
-
-        // Locals
-        // SUBROUTINE ARGUMENT DEFINITIONS:
-
-        // SUBROUTINE PARAMETER DEFINITIONS:
-        // na
-
-        // INTERFACE BLOCK SPECIFICATIONS
-        // na
-
-        // DERIVED TYPE DEFINITIONS
-        // na
-
         // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
         int WaterInletNode;
         int WaterOutletNode;
-        auto &HWBaseboard = state.dataHWBaseboardRad->HWBaseboard;
-        auto &LastSysTimeElapsed = state.dataHWBaseboardRad->LastSysTimeElapsed;
-        auto &QBBRadSrcAvg = state.dataHWBaseboardRad->QBBRadSrcAvg;
-        auto &LastQBBRadSrc = state.dataHWBaseboardRad->LastQBBRadSrc;
-        auto &QBBRadSource = state.dataHWBaseboardRad->QBBRadSource;
-        auto &LastTimeStepSys = state.dataHWBaseboardRad->LastTimeStepSys;
+        auto const &hWBaseboard = state.dataHWBaseboardRad->HWBaseboard(BaseboardNum);
+        auto &lastSysTimeElapsed = state.dataHWBaseboardRad->LastSysTimeElapsed(BaseboardNum);
```

Comment:
All of these are bad uses of local references. If the variable is read only, then copy by value. If the value is written, don't use a reference at all. Also, they should not be `auto`.

Reply:
I'm confused. These were changed to be references to objects. How could these not be `auto`? I can certainly replace `lastSysTimeElapsed` with `state.dataHWBaseboardRad->LastSysTimeElapsed(BaseboardNum)` but I don't see what's wrong with what is here now.

Reply:
The array `state.dataHWBaseboardRad->LastSysTimeElapsed` is not an array of objects, it's an array of `Real64`'s. And so a reference to any element in it is a reference to a `Real64`, not to an object.

Reply:
Different question, why is `LastSysTimeElapsed` a separate array and not part of `HWBaseboard`?

Reply:
You can ask that question about 50% of the arrays in EnergyPlus.